### ISSUE TITLE
Shuffle history

### ISSUE THREAD

**Issue by toto789** [quodlibet/quodlibet#4465]
The proposed change
-------------------
Perhaps a simple button-icon that would display the last song(s) played (ideally with the possibility of clicking on it to replay it, or, with a right click, displaying the same context menu as on a song title).

Benefits of this change
-----------------------
Allow us to know what has just been played (we want to know who this superb piece is from, in a very large music library of which we do not know everything perfectly, but we are not next to the computer, we need several seconds to wake up the screen, and when you are there Quod Libet has already moved on to another song).

Any issues this might cause
---------------------------
No idea.

**Comment by declension** [comment]
See #3160  #2594 #1407

**Comment by toto789** [comment]
I don't mean songs in queue but in a playlist or in a selection in other view.
I read _the only option AFAIK to find the last played songs, is to run the default search "Recently played"_ in #1890, but I can't find such an option in QL 4.7.

**Comment by nemethf** [comment]
One possibility is to customize the headers by adding a new column with a tag expression of `~#lastplayed` and then sort the track list by this column.

The other is to scrobble the played tracks to listenbrainz or a local maloja server and check the history there.

Finally, I have a little events plugin that maintains a playlist with recently played tracks.  If there is interest in it, I can contribute it to quodlibet.

**Comment by Denebet** [comment]
Good point! But I now use gmusicbrowser which, in addition to its fantastic weighted random system, supports the history of the last songs played.
